test/coinselector.spec.ts

```diff
@@ -0,0 +1,92 @@
+            expectedCoinCount: 10, // All the coins are going to be selected.
+        },
+        {
+            testName: 'change should be considered when greater than dust',
+            testCoinValues: [5000, 4000],
+            txOutValue: 1000, // Less output so that the change is greater than dust.
+            expectedChange: 2185, // expectedChange = totalSelectedCoin - (txOutValue + transactionFees).
+            expectedCoinCount: 1, // One Coin will suffice for a 1000 sats output for this case.
+        },
+    ])(
+        '%s',
+        ({ txOutValue, testCoinValues, expectedChange, expectedCoinCount }) => {
+            const coins: Coin[] = [];
+            let totalBalance: number = 0;
+
+            // Calculate the total balance based on the fixed array of coin values.
+            for (const value of testCoinValues) {
+                const coin = new Coin({ value });
+                coins.push(coin);
+                // Calculate the total balance by subtracting the spending fee.
+                totalBalance += coin.value - coin.estimateSpendingFee(feeRate);
+            }
+            
+            // Create a new Transaction
+            const transaction: Transaction = new Transaction();
+
+            // Fixed fees for the transaction calculated based on the total size of the transaction and the change output size.
+            // transactionFees = (virtualSize + changeOutputSize) * feeRate, P2WPKH output is 31 bytes.
+            const transactionFees = 250;
+
+            // txOutValue is not defined for the first test in order to demonstrate its calculation.
+            if (!txOutValue) {
+                // Fixed transaction output value that ensures the cost of change is dust.
+                txOutValue =
+                    totalBalance -
+                    transactionFees -
+                    coinSelector.costOfChange +
+                    1;
+            }
+
+            // Set the transaction output value with a random output script.
+            // Random hex string of 20 bytes and `0014${genRanHex}` will be a valid P2WPKH script.
+            const genRanHex = (size: number) =>
+                [...Array(size)]
+                    .map(() => Math.floor(Math.random() * 16).toString(16))
+                    .join('');
+            transaction.addOutput(
+                Buffer.from(`0014${genRanHex(40)}`), // Ensure that the first parameter passed to addOutput is a Buffer.
+                txOutValue,
+            );
+
+            // Select coins and change using the CoinSelector.
+            const { coins: selectedCoins, change } = coinSelector.select(
+                coins,
+                transaction,
+            );
+
+            // Assert that the selected coins' size meets the expected count.
+            expect(selectedCoins.length).toBe(expectedCoinCount);
+
+            // Assert the change against the expected change condition.
+            expect(change).toBe(expectedChange);
```

Comment:
nit: remove empty line
```suggestion
            // Assert that the selected coins' size meets the expected count.
            expect(selectedCoins.length).toBe(expectedCoinCount);
            // Assert the change against the expected change condition.
            expect(change).toBe(expectedChange);
```